server/controllers/refer/referralsController.test.ts

```diff
@@ -388,4 +388,23 @@ describe('ReferralsController', () => {
       })
     })
   })
+
+  describe('reason', () => {
+    it('renders the reason for referral page', async () => {
+      const person = personFactory.build()
+      personService.getPerson.mockResolvedValue(person)
+
+      const referral = referralFactory.build({ offeringId: courseOffering.id, prisonNumber: person.prisonNumber })
+      referralService.getReferral.mockResolvedValue(referral)
+
+      const requestHandler = referralsController.reason()
+      await requestHandler(request, response, next)
+
+      expect(response.render).toHaveBeenCalledWith('referrals/reason', {
+        pageHeading: 'Add reason for referral and supporting information',
+        person,
+        referral,
+      })
+    })
+  })
```

Comment:
I missed this when reviewing other pages, but we should have a ``describe('when the person service returns `null`'`` block here